src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentCmdletWithParameters.cs

```diff
@@ -0,0 +1,147 @@
+        [ValidateNotNull]
+        public override Hashtable TemplateObject { get; set; }
+
+        [Parameter(ParameterSetName = ResourceGroupParameterSetWithTemplateFileParameterObject,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Uri or local path to the template file.")]
+        [Parameter(ParameterSetName = ResourceGroupParameterSetWithTemplateFileParameterFile,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Uri or local path to the template file.")]
+        [Parameter(ParameterSetName = ResourceGroupParameterSetWithParameterlessTemplateFile,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Uri or local path to the template file.")]
+        [Parameter(ParameterSetName = SubscriptionParameterSetWithTemplateFileParameterObject,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Uri or local path to the template file.")]
+        [Parameter(ParameterSetName = SubscriptionParameterSetWithTemplateFileParameterFile,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Uri or local path to the template file.")]
+        [Parameter(ParameterSetName = SubscriptionParameterSetWithParameterlessTemplateFile,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Uri or local path to the template file.")]
+        [Parameter(ParameterSetName = ManagementGroupParameterSetWithTemplateFileParameterObject,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Uri or local path to the template file.")]
+        [Parameter(ParameterSetName = ManagementGroupParameterSetWithTemplateFileParameterFile,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Uri or local path to the template file.")]
+        [Parameter(ParameterSetName = ManagementGroupParameterSetWithParameterlessTemplateFile,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Uri or local path to the template file.")]
+        [Parameter(ParameterSetName = TenantParameterSetWithTemplateFileParameterObject,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Uri or local path to the template file.")]
+        [Parameter(ParameterSetName = TenantParameterSetWithTemplateFileParameterFile,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Uri or local path to the template file.")]
+        [Parameter(ParameterSetName = TenantParameterSetWithParameterlessTemplateFile,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Uri or local path to the template file.")]
+        [ValidateNotNullOrEmpty]
+        [Alias("TemplateUri")]
+        public override string TemplateFile { get; set; }
```

Comment:
>TemplateFile [](start = 31, length = 12)

I checked TemplateFile appears in the name of each parameter set name constant and it seems to be (same for the other permuations), but this stuff is pretty much write-only.

Is the existing test coverage good enough to confirm whether we're going to cause any back compat issues? (I don't see an issue, but who knows?)

Reply:
I think we have good coverage on New-AzResourceGroupDeployment, but not for New-AzDeployment. I'll see if I can add some tests there.